### PR TITLE
Fix for issue #12673: On wide screens, the text in Learn > Classes is too wide to be comfortably readable

### PR DESCRIPTION
**Pull request by octopols**
Resolves: #12673 

The fix is working as intended, but further improvements are needed for the ClassPage.qml file.
Updated PR for #16087 

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

File: src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml [hunk 1]
```diff
@@ -61,13 +63,13 @@ FocusScope {
...
 
     Rectangle {
-        anchors.top: parent.top
         anchors.left: parent.left
         anchors.leftMargin: root.sideMargin
         anchors.right: parent.right
         anchors.rightMargin: root.sideMargin
 
         height: authorInfo.height
+        width: Math.min(1000, parent.width - root.sideMargin)
```
Comment (cbjeukendrup) [inline src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:72]: I'm afraid this conflicts with `anchors.right: parent.right`. Normally, you either specify the width, or two anchors. So in this case, you'll want to remove `anchors.right`.

Also, I think you should subtract `2 * root.sideMargin`, namely one for the left margin and one for the right margin.

File: src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml [hunk 2]
```diff
@@ -27,8 +27,10 @@ import QtGraphicalEffects 1.0
 import MuseScore.Ui 1.0
 import MuseScore.UiComponents 1.0
 
-FocusScope {
+StyledFlickable {
     id: root
+    contentWidth: parent.width
+    contentHeight: authorInfo.height + sideMargin
```
Comment (cbjeukendrup) [inline src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:33]: Our style is usually to put new property declarations at the top, and property assignments below that. That would mean moving these two properties below `property int sideMargin: …`.

File: src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml (hunk 1 above)
Comment (octopols) [reply src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:72]: Should I remove `anchors.rightMargin` as well? I think the width logic is enough to set the right margin automatically.

Comment (cbjeukendrup) [reply src/learn/qml/MuseScore/Learn/internal/ClassesPage.qml:72]: Yes, `anchors.rightMargin` without `anchors.right` has no effect.

**Comment by cbjeukendrup** [comment]
@octopols Could you please rebase this PR once more? That should make the VTests happy :)

**Comment by octopols** [comment]
> @octopols Could you please rebase this PR once more? That should make the VTests happy :)

Sure

**Review by cbjeukendrup — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**